### DISCUSSION TITLE
Question about `utils/oof.pkl` and request for ROC cure about every model of fit()

### DISCUSSION THREAD

**Discussion by Zjianglin** [autogluon/autogluon#4329] [Q&A]
Hi developers, thanks for your excellent work!

I have a small dataset (`127` samples X `1024` features). The `label` is protein activity (positive class `active` vs. negative class `inactive`). I used `AutoGluon` to `fit` a best model to predict some other homologs.
Here is my core codes:
```python
labelx = 'activity'
train_data = pd.DataFrame(data=Xfeatures0)
train_data.loc[:, labelx] = ylabels0
train_data.index = seq_names1d
predictorx = TabularPredictor(label=labelx, path=model_fpx, eval_metric='f1', learner_kwargs={'positive_class': 'active'}).fit( train_data, presets=['best_quality'], num_stack_levels=1, num_bag_folds=5, num_bag_sets=5, time_limit=limit_secs, num_gpus=4,  verbosity=3)
```
Here is head running output:
```
Verbosity: 3 (Detailed Logging)
=================== System Info ===================
AutoGluon Version:  1.1.1
Python Version:     3.10.14
Operating System:   Linux
Platform Machine:   x86_64
Platform Version:   #1 SMP Tue Mar 31 23:36:51 UTC 2020
CPU Count:          48
GPU Count:          4
Memory Avail:       169.88 GB / 187.58 GB (90.6%)
Disk Space Avail:   264064.31 GB / 531106.51 GB (49.7%)
===================================================
Presets specified: ['best_quality']
============ fit kwarg info ============
User Specified kwargs:
{'auto_stack': True,
 'num_bag_folds': 5,
 'num_bag_sets': 5,
 'num_stack_levels': 1,
 'verbosity': 3}
Full kwargs:
{'_feature_generator_kwargs': None,
 '_save_bag_folds': None,
 'ag_args': None,
 'ag_args_ensemble': None,
 'ag_args_fit': None,
 'auto_stack': True,
 'calibrate': 'auto',
 'ds_args': {'clean_up_fits': True,
             'detection_time_frac': 0.25,
             'enable_ray_logging': True,
             'holdout_data': None,
             'holdout_frac': 0.1111111111111111,
             'memory_safe_fits': True,
             'n_folds': 2,
             'n_repeats': 1,
             'validation_procedure': 'holdout'},
 'excluded_model_types': None,
 'feature_generator': 'auto',
 'feature_prune_kwargs': None,
 'holdout_frac': None,
 'hyperparameter_tune_kwargs': None,
 'included_model_types': None,
 'keep_only_best': False,
 'name_suffix': None,
 'num_bag_folds': 5,
 'num_bag_sets': 5,
 'num_stack_levels': 1,
 'pseudo_data': None,
 'refit_full': False,
 'save_bag_folds': None,
 'save_space': False,
 'set_best_to_refit_full': False,
 'unlabeled_data': None,
 'use_bag_holdout': False,
 'verbosity': 3}
========================================
Setting dynamic_stacking from 'auto' to True. Reason: Enable dynamic_stacking when use_bag_holdout is disabled. (use_bag_holdout=False)
Stack configuration (auto_stack=True): num_stack_levels=1, num_bag_folds=5, num_bag_sets=5
DyStack is enabled (dynamic_stacking=True). AutoGluon will try to determine whether the input data is affected by stacked overfitting and enable or disable stacking as a consequence.
        This is used to identify the optimal `num_stack_levels` value. Copies of AutoGluon will be fit on subsets of the data. Then holdout validation data is used to detect stacked overfitting.
        Running DyStack for up to 10800s of the 43200s of remaining time (25%).
Running DyStack sub-fit in a ray process to avoid memory leakage. Enabling ray logging (enable_ray_logging=True). Specify `ds_args={'enable_ray_logging': False}` if you experience logging issues.
2024-07-14 13:07:52,624 INFO worker.py:1743 -- Started a local Ray instance. View the dashboard at ESC[1mESC[32m127.0.0.1:8265 ESC[39mESC[22m
                Context path: "//Eminus10fitertedV2/0stats/sklearn/fastBest12gpu/autogluon_240628_A_activityKnown_ProtTransT5UniRef50_active_vs_inactive_fastBest12gpu/ds_sub_fit/sub_fit_ho"
ESC[36m(_dystack pid=59259)ESC[0m Running DyStack sub-fit ...
ESC[36m(_dystack pid=59259)ESC[0m Saving Eminus10fitertedV2/0stats/sklearn/fastBest12gpu/autogluon_240628_A_activityKnown_ProtTransT5UniRef50_active_vs_inactive_fastBest12gpu/ds_sub_fit/sub_fit_ho/learner.pkl
ESC[36m(_dystack pid=59259)ESC[0m Saving Eminus10fitertedV2/0stats/sklearn/fastBest12gpu/autogluon_240628_A_activityKnown_ProtTransT5UniRef50_active_vs_inactive_fastBest12gpu/ds_sub_fit/sub_fit_ho/predictor.pkl
SC[36m(_dystack pid=59259)ESC[0m Train Data Rows:    112
ESC[36m(_dystack pid=59259)ESC[0m Train Data Columns: 1024
ESC[36m(_dystack pid=59259)ESC[0m Label Column:       activity
ESC[36m(_dystack pid=59259)ESC[0m Problem Type:       binary
ESC[36m(_dystack pid=59259)ESC[0m Preprocessing data ...
ESC[36m(_dystack pid=59259)ESC[0m Selected class <--> label mapping:  class 1 = active, class 0 = inactive
ESC[36m(_dystack pid=59259)ESC[0m Using Feature Generators to preprocess the data ...
ESC[36m(_dystack pid=59259)ESC[0m Fitting AutoMLPipelineFeatureGenerator...
ESC[36m(_dystack pid=59259)ESC[0m       Available Memory:                    172660.58 MB
ESC[36m(_dystack pid=59259)ESC[0m       Train Data (Original)  Memory Usage: 0.44 MB (0.0% of available memory)
ESC[36m(_dystack pid=59259)ESC[0m       Inferring data type of each feature based on column values. Set feature_metadata_in to manually specify special dtypes of the features.
ESC[36m(_dystack pid=59259)ESC[0m       Stage 1 Generators:
ESC[36m(_dystack pid=59259)ESC[0m               Fitting AsTypeFeatureGenerator...
ESC[36m(_dystack pid=59259)ESC[0m                       Original Features (exact raw dtype, raw dtype):
ESC[36m(_dystack pid=59259)ESC[0m                               ('float32', 'float') : 1024 | ['0', '1', '2', '3', '4', ...]
ESC[36m(_dystack pid=59259)ESC[0m                       Types of features in original data (raw dtype, special dtypes):
ESC[36m(_dystack pid=59259)ESC[0m                               ('float', []) : 1024 | ['0', '1', '2', '3', '4', ...]
ESC[36m(_dystack pid=59259)ESC[0m                       Types of features in processed data (raw dtype, special dtypes):
ESC[36m(_dystack pid=59259)ESC[0m                               ('float', []) : 1024 | ['0', '1', '2', '3', '4', ...]
ESC[36m(_dystack pid=59259)ESC[0m                       0.1s = Fit runtime
ESC[36m(_dystack pid=59259)ESC[0m                       1024 features in original data used to generate 1024 features in processed data
```

After `.fit()`, the best model is `WeightedEnsemble_L3`,  due to my small size training dataset, I want to get a ROC curve to evaluate the model whether is overfiting or not? How could I do that? What is the  file contnet in `models/WeightedEnsemble_L3/utils/oof.pkl` , could I get the ROC from that? thanks.
